version/version.go

```diff
@@ -64,6 +71,58 @@ func (v *Version) AtLeast(minVersion string) bool {
 	return v.Compare(minVersion) <= 0
 }
 
+// Check if candidate is an upgrade/downgrade of the base version only in the specified position
+// which corresponds to Major, Minor and Patch positions.
+// If True, returns the version, if false return nil.
+// Examples: currVersion = Version{"1.2.1}
+// currVersion.IsUpgradeAtPosition(Version{"1.2.2"}, 3) => Version{"1.2.2"}.
+// currVersion.IsUpgradeAtPosition(Version{"1.2.0"}, 3) => nil.
+// currVersion.IsUpgradeAtPosition(Version{"2.2.0"}, 1) => Version{"2.2.0"}.
+func (v *Version) IsUpgradeAtPosition(candidate Version, semanticPosition int) (*Version, error) {
```

Comment:
I think that we can make this utility more "standard" and easier to comprehend by implementing the following changes to the API:
1. This utility should be receive a type named 'VersionPosition' as the second argument, instead of an 'int'.
2. semanticPosition --> versionPosition 
3. IsUpgradeAtPosition --> CompareAtPosition
4. The function should return either -1, 0 or 1 according to the comparison result.